snuba/datasets/plans/query_plan.py

```diff
@@ -33,7 +33,8 @@ class ClickhouseQueryPlan:
     """
 
     query: Query
-    plan_processors: Sequence[QueryProcessor]
+    plan_query_processors: Sequence[QueryProcessor]
+    db_query_processors: Sequence[QueryProcessor]
```

Comment:
This did not exist before because there was never a case for executing the db query processor outside an execution strategy so db query processors were directly passed to the strategy. Now the split between these two is still required (when we run a query on an individual entity), but when we run a composite query, the query execution strategy is not provided by the storage anymore. At best it would be provided by a dataset, so we skip that step and we just need to run all processors on every subquery as if they were not split (in fact they are not, it is the main composite query that may be split).